web-template/src/test/java/org/ehrbase/webtemplate/interpreter/InterpreterTest.java

```diff
@@ -0,0 +1,432 @@
+/*
+ * Copyright (c) 2022 vitasystems GmbH and Hannover Medical School.
+ *
+ * This file is part of project openEHR_SDK
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehrbase.webtemplate.interpreter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.assertj.core.groups.Tuple;
+import org.ehrbase.aql.dto.AqlDto;
+import org.ehrbase.aql.dto.containment.Containment;
+import org.ehrbase.aql.dto.containment.ContainmentDto;
+import org.ehrbase.aql.dto.path.AqlPath;
+import org.ehrbase.aql.dto.path.predicate.PredicateHelper;
+import org.ehrbase.aql.dto.path.predicate.PredicateLogicalAndOperation;
+import org.ehrbase.aql.dto.select.SelectFieldDto;
+import org.ehrbase.aql.parser.AqlToDtoParser;
+import org.ehrbase.test_data.operationaltemplate.OperationalTemplateTestData;
+import org.ehrbase.webtemplate.model.WebTemplateNode;
+import org.ehrbase.webtemplate.templateprovider.TestDataTemplateProvider;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+/**
+ * @author Stefan Spiska
+ */
+class InterpreterTest {
```

Comment:
Having a small test case using select fields from a where condition would be helpful for descriptive purposes.